### DISCUSSION TITLE
Error with md-roam: Debugger entered--Lisp error: (error "rx ‘**’ range error")

### DISCUSSION THREAD

**Discussion by rickswe** [company-mode/company-mode#1449] [Q&A]
Hello,

I have been getting an errors with md-roam/org-roam-ui. When I disable company
the error goes away. This usually happens when an org commands are being run
in a non org buffer which was discussed in https://github.com/nobiot/md-roam/issues/81.
It happens when I start editing in a markdown file. 

```lisp
Debugger entered--Lisp error: (error "rx ‘**’ range error")
  error("rx `%s' range error" **)
  rx--translate-bounded-repetition(** (1 0 "*"))
  rx--translate-**((1 0 "*"))
  rx--translate-form((** 1 0 "*"))
  rx--translate((** 1 0 "*"))
  rx--translate-seq((line-start (** 1 0 "*") " "))
  rx--translate-form((seq line-start (** 1 0 "*") " "))
  rx--translate((seq line-start (** 1 0 "*") " "))
  rx-to-string((seq line-start (** 1 0 "*") " "))
  org-element-headline-parser(nil fast)
  org-element--parse-to(379)
  org-element-at-point()
  org-in-src-block-p()
  org-roam-complete-everywhere()
  completion--capf-wrapper(org-roam-complete-everywhere optimist)
  company--capf-wrapper(org-roam-complete-everywhere optimist)
  company--capf-data-real()
  company--capf-data()
  company-capf(prefix)
  company--force-sync(company-capf (prefix) company-capf)
  company--multi-prefix((company-yasnippet company-files company-bbdb company-semantic company-cmake company-capf))
  company--multi-backend-adapter((:separate company-yasnippet company-files company-bbdb company-semantic company-cmake company-capf :with company-tempo company-clang company-dabbrev-code company-gtags company-keywords company-oddmuse company-dabbrev) prefix)
  apply(company--multi-backend-adapter (:separate company-yasnippet company-files company-bbdb company-semantic company-cmake company-capf :with company-tempo company-clang company-dabbrev-code company-gtags company-keywords company-oddmuse company-dabbrev) prefix)
  company-call-backend-raw(prefix)
  company--force-sync(company-call-backend-raw (prefix) (:separate company-yasnippet company-files company-bbdb company-semantic company-cmake company-capf :with company-tempo company-clang company-dabbrev-code company-gtags company-keywords company-oddmuse company-dabbrev))
  company-call-backend(prefix)
  company--continue()
  company--perform()
  company-post-command()
```

Warnings

```lisp
⛔ Warning (org-element): ‘org-element-at-point’ cannot be used in non-Org buffer #<buffer fake.md> (markdown-mode)
⛔ Warning (org-element): org-element--cache: Org parser error in fake.md::369. Resetting.
 The error was: (error "rx ‘**’ range error")
 Backtrace:
nil
 Please report this to Org mode mailing list (M-x org-submit-bug-report).
⛔ Warning (org-element): ‘org-element-at-point’ cannot be used in non-Org buffer #<buffer fake.md> (markdown-mode)
⛔ Warning (org-element): org-element--cache: Org parser error in fake.md::370. Resetting.
 The error was: (error "rx ‘**’ range error")
 Backtrace:
nil
 Please report this to Org mode mailing list (M-x org-submit-bug-report).
⛔ Warning (org-element): ‘org-element-at-point’ cannot be used in non-Org buffer #<buffer fake.md> (markdown-mode)
⛔ Warning (org-element): org-element--cache: Org parser error in fake.md::371. Resetting.
 The error was: (error "rx ‘**’ range error")
 Backtrace:
nil
 Please report this to Org mode mailing list (M-x org-submit-bug-report).
⛔ Warning (org-element): ‘org-element-at-point’ cannot be used in non-Org buffer #<buffer fake.md> (markdown-mode)
```

Thanks

**Comment by dgutov** [comment]
Apparently `org-roam-complete-everywhere` is in `completion-at-point-functions`? And the error says that it shouldn't be used in non-Org buffers. Seems like a configuration or design error somewhere in `org-roam` (sorry, I'm not familiar with what that package does).

If instead of having company-mode enabled, you press `C-M-i` in exact same consequences (after "org commands are being run
in a non org buffer" as you said) to trigger the default completion UI, do you see the same error?

**Comment by rickswe** [comment]
>Apparently org-roam-complete-everywhere is in completion-at-point-functions? 
I'm not sure, my elisp knowlege  is limited. I know enough to get by. 

I get no errors when I disable company mode and complete with C-M-i but it only
completes basic words. Not my yasnippets. 

C-M-i runs the command ispell-complete-word (found in markdown-mode-map), which is an
autoloaded interactive native-compiled Lisp function in ‘ispell.el’.

**Reply by dgutov** [reply]
In that case, you can try `M-x completion-at-point`.

**Reply by rickswe** [reply]
It looks like the error is coming from org-roam-complete-everywhere. What I did not understand is
why enabling and disabling company was giving me the errors?

```elisp
Debugger entered--Lisp error: (error "rx ‘**’ range error")
  error("rx `%s' range error" **)
  rx--translate-bounded-repetition(** (1 0 "*"))
  rx--translate-**((1 0 "*"))
  rx--translate-form((** 1 0 "*"))
  rx--translate((** 1 0 "*"))
  rx--translate-seq((line-start (** 1 0 "*") " "))
  rx--translate-form((seq line-start (** 1 0 "*") " "))
  rx--translate((seq line-start (** 1 0 "*") " "))
  rx-to-string((seq line-start (** 1 0 "*") " "))
  org-element-headline-parser(nil fast)
  org-element--parse-to(412)
  org-element-at-point()
  org-in-src-block-p()
  org-roam-complete-everywhere()
  completion--capf-wrapper(org-roam-complete-everywhere all)
  completion-at-point()
  funcall-interactively(completion-at-point)
  command-execute(completion-at-point record)
  execute-extended-command(nil "completion-at-point" "com")
  funcall-interactively(execute-extended-command nil "completion-at-point" "com")
  command-execute(execute-extended-command)
```

**Reply by dgutov** [reply]
Because company initiates completion automatically?

You can disable that with `(setq company-idle-delay nil)`.

**Reply by rickswe** [reply]
I'm not 100% sure what the issue is but for now I can use company-mode
by setting the variable `org-roam-completion-everywhere` to nil :)

**Reply by dgutov** [reply]
Perhaps you should report that to org-roam.

The fact that `M-x completion-at-point` results in an error seems to indicate that `org-roam-completion-everywhere` doesn't work as intended.

**Reply by rickswe** [reply]
You're right, this all got started in [md-roam issue 81](https://github.com/nobiot/md-roam/issues/81). I hope
they don't tell me that org-roam  is only intended to be used with org files but it's worth a shot. 

What [md-roam](https://github.com/nobiot/md-roam) does is it enables the user to use markdown files with [org-roam](https://github.com/org-roam/org-roam). This is convenient because a user can use [org-roam-ui](https://github.com/org-roam/org-roam-ui) to display org files and markdown files in a graph (personal knowledge management system.).
